docs/marionette.view.md

```diff
@@ -569,84 +628,99 @@ By prefixing with `@ui`, we can change the underlying template without having to
-    'click .button': 'onClickButton'
-  },
+The onEvent listeners follow clearly defined rules for determining whether they
+should be called:
 
-  // Triggers hash converts DOM events directly to view events catchable on the parent.
-  triggers: {
-    'submit form': 'submit:form'
-  },
+  1. Split the event name around `:`
+  2. Capitalize the first letter of each "word"
+  3. Prepend `on` to the event name
+  4. If the associated method exists, call it
+  5. Any extra arguments passed to `triggerMethod` get passed in
 
-  onClickButton: function () {
-    this.triggerMethod('show:message', 'foo');
-  }
-});
+A few simple examples:
 
-// The parent uses childEvents to catch that custom event on the child view
-var ParentView = Marionette.View.extend({
+  1. `triggerMethod('render')` calls `onRender()`
+  2. `triggerMethod('before:render')` calls `onBeforeRender()`
+  3. `triggerMethod('before:sync', model)` calls `onBeforeSync(model)`
 
-  childEvents: {
-    'show:message': 'onChildShowMessage',
-    'submit:form': 'onChildSubmitForm'
-  },
+### Lifecycle Events
 
-  onChildShowMessage: function (childView, message) {
-    console.log('A child view fired show:message with ' + message);
-  },
-  // Methods called from the triggers hash do not have access to DOM events
-  // Any logic requiring the original DOM event should be handled in it's respective view
-  onChildSubmitForm: function (childView) {
-    console.log('A child view fired submit:form');
-  }
-});
-```
+Marionette views defined a number of events during the creation and destruction
+lifecycle - when the view is displayed in and emptied from a region. In the
+documentation, we will reference the event name, however every event here
+respect the [rules for onEvent listeners](rules-for-onevent-listeners).
+
+#### View Creation Lifecycle
+
+When a view is initialized and then displayed inside a region (using
+`showChildView()`) a set of events will be called in a specific order.
+
+| Order |      Event      |
+| :---: |-----------------|
+|   1   | `before:render` |
+|   2   | `render`        |
+|   3   | `before:attach` |
+|   4   | `attach`        |
+|   5   | `dom:refresh`   |
 
-## Events and Callback Methods
+#### View Destruction Lifecycle
 
-There are several events and callback methods that are called
-for an View. These events and methods are triggered with the
-[Marionette.triggerMethod](./marionette.functions.md#marionettetriggermethod)
-function which triggers the event and a corresponding `on{EventName}` method.
+When  `region.empty()` is called, the view will be destroyed, calling events as
+part of the destruction lifecycle.
 
-### `before:render` / `onBeforeRender` event
+| Order |       Event       |
+| :---: |-------------------|
+|   1   |  `before:destroy` |
+|   2   |  `before:detach`  |
+|   3   |  `detach`         |
+|   4   |  `destroy`        |
+
+#### View Creation Events
+
+These events are fired during the view's creation and rendering in a region.
```

Comment:
I wouldn't have guessed this to be the order
I would have thought

1. before:detach
2. detach
3. before:destroy
4. destroy